### DISCUSSION TITLE
Pygbag Index Template

### DISCUSSION THREAD

**Discussion by Avashist1998** [pygame-web/pygbag#87] [Q&A] [answered]
# Pygbag Index Template

I want to start of by saying that I really appreciate the Pygbag team as done. I recently build a simple game in pygame and used Pygbag to distribute it using Github Pages. I am very happy with the tool and worked well after a little bit of struggling and couple visits to the docs. I am sorry if these question can be easily found in the docs. 

## Questions:
- I want the game website to have static content complimenting the game. How would I got about achieving that?
- Can I create a HTML file and add a tag in which the game will be embedded during build?
- When I first load the page, I need to click the screen for the game to start. Is there a way to skip the clicking step?
- Closing the tab requires confirmation, how can that be skipped?

Links:
[Maze Game](https://avashist1998.github.io/maze_game/)
[Maze Game Repo](https://github.com/Avashist1998/maze_game/)

Thank you so much for making a wonderful tool.

**Comment by pmp-p** [comment, marked as answer]
Hi , thanks for the praise :)

- the template option that use the default.tmpl can mix both python and html
- if you don't have music on start you can prepend `--ume_block 0` option on command line
- closing tab can be avoided from python by nulling the handler `platform.window.onbeforeunload` at runtime or by prepending`--can_close 1` on command line

**Comment by Avashist1998** [comment]
Thank you so much for the response. I have tested that flags/args proposed, and they lead to the desired behavior.
 
On the matter of using the custom default.tmpl file for custom index.html page generation.

Major concerns with custom tmpl file
- `Pygbag` version updates may lead to breaking changes in the future.
- (Intertwined with Pygbag library) It requires a deeper understanding of `Pygbag`, which not ideal from users perspective.
- Limits the template to tmpl file type, which is not generated by most JS compilers and libraries.

Suggestion (What are your thoughts?)
User creates a shell HTML with `Pygbag` tag and `Pygbag` ejected injects the tags with the appropriate HTML, JavaScript, and wasm during the build. 
- (More flexibility) Independence  JS libs to generate the HTML

I may be misunderstanding your suggestion, if not just something to consider.

**Comment by pmp-p** [comment]
Templates don't use pygbag but builtins modules of python runtime that are unlikely to change ( they are in C, come from Panda3D and map emscripten API ), sure there are facilities in pythonrc.py for using them but you could put everything in the template instead : it's just more work for low level things that are unlikely to change before v1.

But indeed the goal of pygbag is to allow running cross platform games on web almost "out of the box" , so using html/js injection - is easy - but probably toxic for game portability on long term. For making web-only games there are probably better alternatives in javascript/wasm world. So i would not advise to use injection when it can be avoided.

**Reply by Avashist1998** [reply]
Thanks
